crates/wasm/src/planner.rs

```diff
@@ -135,6 +147,22 @@ impl<R: RngCore + CryptoRng> Planner<R> {
         self
     }
 
+    /// Calculate gas cost-based fees and add to the transaction plan.
+    ///
+    /// This function should be called once.
+    pub fn add_gas_fees(&mut self) -> &mut Self {
```

Comment:
Instead of having this be a separate call that needs to be done at the end, why not put it inside `set_gas_prices` and it gets done as someone adds the gas price?

Reply:
`add_gas_fees` depends on the actions that have been added to the plan. We must ensure that new actions cannot be added after `add_gas_fees() `is executed, so it is better to do it once at the end. And yes, this solution is also inherited from pcli